diff --git a/actors/enemies/slime/slime.tscn b/actors/enemies/slime/slime.tscn
index 63ccfb3..f065a37 100644
--- a/actors/enemies/slime/slime.tscn
+++ b/actors/enemies/slime/slime.tscn
@@ -1,23 +1,23 @@
 [gd_scene format=3 uid="uid://b2aypx00i23wq"]
 
-[ext_resource type="Texture2D" uid="uid://bifnm001bqeaw" path="res://player/Sprites/Shadow.png" id="1_j284o"]
-[ext_resource type="Script" uid="uid://dgwho80w0bwf8" path="res://enemies/slime/slime.gd" id="1_yu3x6"]
-[ext_resource type="Texture2D" uid="uid://bbvw8ixq8x24c" path="res://enemies/slime/Slime.png" id="2_6326o"]
-[ext_resource type="Resource" uid="uid://ct8o2kpbu3g55" path="res://enemies/slime/slime_stats.tres" id="2_okhmd"]
-[ext_resource type="PackedScene" uid="uid://b7e5y80fvd4ia" path="res://generalnodes/hurtbox/hurt_box.tscn" id="2_v3fw7"]
-[ext_resource type="PackedScene" uid="uid://w5gedwn5m48d" path="res://generalnodes/hitbox/hit_box.tscn" id="3_0oddt"]
-[ext_resource type="Script" uid="uid://2ohc36m0d0x4" path="res://enemies/scripts/enemy_state_machine.gd" id="4_57eei"]
-[ext_resource type="Script" uid="uid://bgoktucdsh87j" path="res://enemies/scripts/states/enemy_state_idle.gd" id="5_2ui2u"]
-[ext_resource type="Script" uid="uid://3oq1eouo0dne" path="res://enemies/scripts/states/enemy_state_wander.gd" id="6_bkm6f"]
-[ext_resource type="AudioStream" uid="uid://flhgcqdf4v04" path="res://enemies/slime/hit_01.wav" id="6_klbxu"]
-[ext_resource type="AudioStream" uid="uid://dduin2bnc1h85" path="res://enemies/slime/hit_00.wav" id="7_mwtj1"]
-[ext_resource type="Script" uid="uid://cutbyswosjvs0" path="res://enemies/scripts/states/enemy_state_stun.gd" id="11_im24l"]
-[ext_resource type="Script" uid="uid://dvpd4wr288jpv" path="res://enemies/scripts/states/enemy_state_destroy.gd" id="12_0oddt"]
-[ext_resource type="Texture2D" uid="uid://bjmnekt8aq4gn" path="res://enemies/slime/DestroySmoke.png" id="12_yu3x6"]
-[ext_resource type="Script" uid="uid://c40kdvh5as01" path="res://enemies/scripts/states/enemy_state_chase.gd" id="13_yu3x6"]
-[ext_resource type="Texture2D" uid="uid://cmmamg8pmr5hf" path="res://assets/healthbarUnder.png" id="14_im24l"]
-[ext_resource type="Script" uid="uid://b1vkyse6bgt6i" path="res://enemies/scripts/health_bar.gd" id="14_klbxu"]
-[ext_resource type="Texture2D" uid="uid://bf6c14mfpspng" path="res://assets/healthbarProgress.png" id="15_yu3x6"]
+[ext_resource type="Texture2D" uid="uid://bifnm001bqeaw" path="res://actors/player/art/Shadow.png" id="1_j284o"]
+[ext_resource type="Script" uid="uid://dgwho80w0bwf8" path="res://actors/enemies/slime/slime.gd" id="1_yu3x6"]
+[ext_resource type="Texture2D" uid="uid://bbvw8ixq8x24c" path="res://actors/enemies/slime/art/Slime.png" id="2_6326o"]
+[ext_resource type="Resource" uid="uid://ct8o2kpbu3g55" path="res://data/enemies/slime_stats.tres" id="2_okhmd"]
+[ext_resource type="PackedScene" uid="uid://b7e5y80fvd4ia" path="res://core/combat/hurt_box.tscn" id="2_v3fw7"]
+[ext_resource type="PackedScene" uid="uid://w5gedwn5m48d" path="res://core/combat/hit_box.tscn" id="3_0oddt"]
+[ext_resource type="Script" uid="uid://2ohc36m0d0x4" path="res://actors/enemies/enemy_state_machine.gd" id="4_57eei"]
+[ext_resource type="Script" uid="uid://bgoktucdsh87j" path="res://actors/enemies/states/enemy_state_idle.gd" id="5_2ui2u"]
+[ext_resource type="Script" uid="uid://3oq1eouo0dne" path="res://actors/enemies/states/enemy_state_wander.gd" id="6_bkm6f"]
+[ext_resource type="AudioStream" uid="uid://flhgcqdf4v04" path="res://actors/enemies/slime/audio/hit_01.wav" id="6_klbxu"]
+[ext_resource type="AudioStream" uid="uid://dduin2bnc1h85" path="res://actors/enemies/slime/audio/hit_00.wav" id="7_mwtj1"]
+[ext_resource type="Script" uid="uid://cutbyswosjvs0" path="res://actors/enemies/states/enemy_state_stun.gd" id="11_im24l"]
+[ext_resource type="Script" uid="uid://dvpd4wr288jpv" path="res://actors/enemies/states/enemy_state_destroy.gd" id="12_0oddt"]
+[ext_resource type="Texture2D" uid="uid://bjmnekt8aq4gn" path="res://actors/enemies/slime/art/DestroySmoke.png" id="12_yu3x6"]
+[ext_resource type="Script" uid="uid://c40kdvh5as01" path="res://actors/enemies/states/enemy_state_chase.gd" id="13_yu3x6"]
+[ext_resource type="Texture2D" uid="uid://cmmamg8pmr5hf" path="res://core/combat/art/healthbar_under.png" id="14_im24l"]
+[ext_resource type="Script" uid="uid://b1vkyse6bgt6i" path="res://actors/enemies/enemy_health_bar.gd" id="14_klbxu"]
+[ext_resource type="Texture2D" uid="uid://bf6c14mfpspng" path="res://core/combat/art/healthbar_progress.png" id="15_yu3x6"]
 
 [sub_resource type="CircleShape2D" id="CircleShape2D_okhmd"]
 radius = 14.0
diff --git a/actors/player/projectiles/arrow.tscn b/actors/player/projectiles/arrow.tscn
index 50ec806..ec60185 100644
--- a/actors/player/projectiles/arrow.tscn
+++ b/actors/player/projectiles/arrow.tscn
@@ -1,7 +1,7 @@
 [gd_scene format=3 uid="uid://cxbl6qc4wi84k"]
 
-[ext_resource type="Script" uid="uid://3msbgf0wsdc8" path="res://player/scripts/arrow.gd" id="1_arrow"]
-[ext_resource type="Texture2D" uid="uid://bgne3a33tpb8n" path="res://assets/Arrow.png" id="2_6ha8w"]
+[ext_resource type="Script" uid="uid://3msbgf0wsdc8" path="res://actors/player/projectiles/arrow.gd" id="1_arrow"]
+[ext_resource type="Texture2D" uid="uid://bgne3a33tpb8n" path="res://actors/player/projectiles/art/arrow.png" id="2_6ha8w"]
 
 [sub_resource type="RectangleShape2D" id="RectangleShape2D_arrow"]
 size = Vector2(14, 4)
diff --git a/core/combat/hit_box.tscn b/core/combat/hit_box.tscn
index 76e545d..240a7ea 100644
--- a/core/combat/hit_box.tscn
+++ b/core/combat/hit_box.tscn
@@ -1,6 +1,6 @@
 [gd_scene format=3 uid="uid://w5gedwn5m48d"]
 
-[ext_resource type="Script" uid="uid://c5bxwnf6kjo7t" path="res://generalnodes/hitbox/hit_box.gd" id="1_bm3cc"]
+[ext_resource type="Script" uid="uid://c5bxwnf6kjo7t" path="res://core/combat/hit_box.gd" id="1_bm3cc"]
 
 [node name="HitBox" type="Area2D" unique_id=1594258603]
 collision_layer = 256
diff --git a/core/combat/hurt_box.tscn b/core/combat/hurt_box.tscn
index 53eec0b..5dee991 100644
--- a/core/combat/hurt_box.tscn
+++ b/core/combat/hurt_box.tscn
@@ -1,6 +1,6 @@
 [gd_scene format=3 uid="uid://b7e5y80fvd4ia"]
 
-[ext_resource type="Script" uid="uid://dvdiykfc8ocn0" path="res://generalnodes/hurtbox/hurt_box.gd" id="1_mwx43"]
+[ext_resource type="Script" uid="uid://dvdiykfc8ocn0" path="res://core/combat/hurt_box.gd" id="1_mwx43"]
 
 [node name="HurtBox" type="Area2D" unique_id=360467044]
 collision_layer = 0
diff --git a/data/enemies/slime_stats.tres b/data/enemies/slime_stats.tres
index cedf54f..bde1eab 100644
--- a/data/enemies/slime_stats.tres
+++ b/data/enemies/slime_stats.tres
@@ -1,6 +1,6 @@
 [gd_resource type="Resource" script_class="EnemyStats" format=3 uid="uid://ct8o2kpbu3g55"]
 
-[ext_resource type="Script" uid="uid://b0qr047oe60bl" path="res://enemies/scripts/enemy_stats.gd" id="1_xeand"]
+[ext_resource type="Script" uid="uid://b0qr047oe60bl" path="res://core/resources/enemy_stats.gd" id="1_xeand"]
 
 [resource]
 script = ExtResource("1_xeand")
diff --git a/data/skills/archer_skill.tres b/data/skills/archer_skill.tres
index a3d53a4..8d9ac69 100644
--- a/data/skills/archer_skill.tres
+++ b/data/skills/archer_skill.tres
@@ -1,6 +1,6 @@
 [gd_resource type="Resource" script_class="SkillData" load_steps=3 format=3]
 
-[ext_resource type="Script" uid="uid://ccpqg0ooqkxty" path="res://player/skill_data.gd" id="1_skill"]
+[ext_resource type="Script" uid="uid://ccpqg0ooqkxty" path="res://core/resources/skill_data.gd" id="1_skill"]
 [ext_resource type="Texture2D" path="res://actors/player/art/skill_archer.svg" id="2_icon"]
 
 [resource]
diff --git a/data/skills/archer_ultimate.tres b/data/skills/archer_ultimate.tres
index 2b2b039..378121d 100644
--- a/data/skills/archer_ultimate.tres
+++ b/data/skills/archer_ultimate.tres
@@ -1,6 +1,6 @@
 [gd_resource type="Resource" script_class="SkillData" load_steps=3 format=3]
 
-[ext_resource type="Script" uid="uid://ccpqg0ooqkxty" path="res://player/skill_data.gd" id="1_skill"]
+[ext_resource type="Script" uid="uid://ccpqg0ooqkxty" path="res://core/resources/skill_data.gd" id="1_skill"]
 [ext_resource type="Texture2D" path="res://actors/player/art/ultimate_archer.svg" id="2_icon"]
 
 [resource]
diff --git a/data/skills/warrior_skill.tres b/data/skills/warrior_skill.tres
index d0b52a4..1b047c6 100644
--- a/data/skills/warrior_skill.tres
+++ b/data/skills/warrior_skill.tres
@@ -1,6 +1,6 @@
 [gd_resource type="Resource" script_class="SkillData" load_steps=3 format=3]
 
-[ext_resource type="Script" uid="uid://ccpqg0ooqkxty" path="res://player/skill_data.gd" id="1_skill"]
+[ext_resource type="Script" uid="uid://ccpqg0ooqkxty" path="res://core/resources/skill_data.gd" id="1_skill"]
 [ext_resource type="Texture2D" path="res://actors/player/art/skill_warrior.svg" id="2_icon"]
 
 [resource]
diff --git a/dungeon/map/map.tscn b/dungeon/map/map.tscn
index a7c19f4..e81ee1e 100644
--- a/dungeon/map/map.tscn
+++ b/dungeon/map/map.tscn
@@ -1,8 +1,8 @@
 [gd_scene format=3 uid="uid://d0tgltkm5w12x"]
 
-[ext_resource type="Script" uid="uid://0n16fxsfron3" path="res://dungeon/map_generator.gd" id="1_ngjdg"]
-[ext_resource type="Script" uid="uid://cf3e1fkyxmaa1" path="res://dungeon/scenes/map.gd" id="1_ymosu"]
-[ext_resource type="Texture2D" uid="uid://hlebndwru55t" path="res://assets/map.png" id="3_y2wp4"]
+[ext_resource type="Script" uid="uid://0n16fxsfron3" path="res://dungeon/generation/map_generator.gd" id="1_ngjdg"]
+[ext_resource type="Script" uid="uid://cf3e1fkyxmaa1" path="res://dungeon/map/map.gd" id="1_ymosu"]
+[ext_resource type="Texture2D" uid="uid://hlebndwru55t" path="res://dungeon/map/art/map_background.png" id="3_y2wp4"]
 
 [node name="Map" type="Node2D" unique_id=1183846057 groups=["Map"]]
 process_mode = 3
diff --git a/dungeon/map/map_line.tscn b/dungeon/map/map_line.tscn
index 677c19d..9e94fef 100644
--- a/dungeon/map/map_line.tscn
+++ b/dungeon/map/map_line.tscn
@@ -1,6 +1,6 @@
 [gd_scene format=3 uid="uid://bw30i2pfo5msk"]
 
-[ext_resource type="Texture2D" uid="uid://bn3tlx14g8tnp" path="res://dungeon/art/line.png" id="1_fem3m"]
+[ext_resource type="Texture2D" uid="uid://bn3tlx14g8tnp" path="res://dungeon/map/art/line.png" id="1_fem3m"]
 
 [node name="MapLine" type="Line2D" unique_id=310631242]
 texture_repeat = 2
diff --git a/dungeon/map/map_room.tscn b/dungeon/map/map_room.tscn
index 44ef0f9..ecd0cd7 100644
--- a/dungeon/map/map_room.tscn
+++ b/dungeon/map/map_room.tscn
@@ -1,7 +1,7 @@
 [gd_scene format=3 uid="uid://djms65jjb33jy"]
 
-[ext_resource type="Script" uid="uid://csnj57je285pf" path="res://dungeon/scenes/map_room.gd" id="1_cuibe"]
-[ext_resource type="Texture2D" uid="uid://wnei4t31yt7w" path="res://dungeon/art/monster_room_icon.png" id="2_cuibe"]
+[ext_resource type="Script" uid="uid://csnj57je285pf" path="res://dungeon/map/map_room.gd" id="1_cuibe"]
+[ext_resource type="Texture2D" uid="uid://wnei4t31yt7w" path="res://dungeon/map/art/monster_room_icon.png" id="2_cuibe"]
 
 [sub_resource type="CircleShape2D" id="CircleShape2D_hl4ru"]
 radius = 8.0
diff --git a/dungeon/rooms/player_spawn.tscn b/dungeon/rooms/player_spawn.tscn
index 1e4ad19..7a89cbf 100644
--- a/dungeon/rooms/player_spawn.tscn
+++ b/dungeon/rooms/player_spawn.tscn
@@ -1,7 +1,7 @@
 [gd_scene format=3 uid="uid://dev804nrhko7n"]
 
-[ext_resource type="Script" uid="uid://dg8wcnmovkded" path="res://generalnodes/playerspawn/player_spawn.gd" id="1_8n7r2"]
-[ext_resource type="Texture2D" uid="uid://dcu4jbhgb8ugj" path="res://player/Sprites/PlayerSprite02.png" id="2_2oql1"]
+[ext_resource type="Script" uid="uid://dg8wcnmovkded" path="res://dungeon/rooms/player_spawn.gd" id="1_8n7r2"]
+[ext_resource type="Texture2D" uid="uid://dcu4jbhgb8ugj" path="res://actors/player/art/PlayerSprite02.png" id="2_2oql1"]
 
 [node name="PlayerSpawn" type="Node2D" unique_id=442468290]
 script = ExtResource("1_8n7r2")
diff --git a/dungeon/rooms/room_monster.tscn b/dungeon/rooms/room_monster.tscn
index b02007e..479c142 100644
--- a/dungeon/rooms/room_monster.tscn
+++ b/dungeon/rooms/room_monster.tscn
@@ -1,10 +1,10 @@
 [gd_scene format=4 uid="uid://m15co7c1hoee"]
 
-[ext_resource type="Script" uid="uid://bisghiseja2qh" path="res://dungeon/scenes/room_monster.gd" id="1_jo2km"]
-[ext_resource type="Texture2D" uid="uid://cmjcjnyoylinu" path="res://dungeon/art/tileset_free_miniwarriors/Water Background color.png" id="2_s67or"]
-[ext_resource type="Script" uid="uid://gv6eu8bp6g6l" path="res://dungeon/walker_generator.gd" id="3_1e1os"]
-[ext_resource type="Texture2D" uid="uid://cekytqlncodmh" path="res://dungeon/art/tileset_free_miniwarriors/Tilemap_color5.png" id="5_4lth7"]
-[ext_resource type="Texture2D" uid="uid://cx5npccdpigqb" path="res://dungeon/art/tileset_free_miniwarriors/Tilemap_color3.png" id="8_dntdx"]
+[ext_resource type="Script" uid="uid://bisghiseja2qh" path="res://dungeon/rooms/room_monster.gd" id="1_jo2km"]
+[ext_resource type="Texture2D" uid="uid://cmjcjnyoylinu" path="res://dungeon/rooms/tilesets/art/Water Background color.png" id="2_s67or"]
+[ext_resource type="Script" uid="uid://gv6eu8bp6g6l" path="res://dungeon/generation/walker_generator.gd" id="3_1e1os"]
+[ext_resource type="Texture2D" uid="uid://cekytqlncodmh" path="res://dungeon/rooms/tilesets/art/Tilemap_color5.png" id="5_4lth7"]
+[ext_resource type="Texture2D" uid="uid://cx5npccdpigqb" path="res://dungeon/rooms/tilesets/art/Tilemap_color3.png" id="8_dntdx"]
 
 [sub_resource type="TileSetAtlasSource" id="TileSetAtlasSource_jo2km"]
 texture = ExtResource("5_4lth7")
diff --git a/dungeon/run.tscn b/dungeon/run.tscn
index 42f3a0f..f59594f 100644
--- a/dungeon/run.tscn
+++ b/dungeon/run.tscn
@@ -1,7 +1,7 @@
 [gd_scene format=3 uid="uid://bsbe7wll3uofc"]
 
-[ext_resource type="Script" uid="uid://bvt8qw3ql1bfy" path="res://generalnodes/dungeon.gd" id="1_286u4"]
-[ext_resource type="PackedScene" uid="uid://d0tgltkm5w12x" path="res://dungeon/scenes/map.tscn" id="2_hruxe"]
+[ext_resource type="Script" uid="uid://bvt8qw3ql1bfy" path="res://dungeon/run.gd" id="1_286u4"]
+[ext_resource type="PackedScene" uid="uid://d0tgltkm5w12x" path="res://dungeon/map/map.tscn" id="2_hruxe"]
 
 [node name="Dungeon" type="Node2D" unique_id=2094591279]
 y_sort_enabled = true
diff --git a/props/plants/plant.tscn b/props/plants/plant.tscn
index 0e7c844..f18e79b 100644
--- a/props/plants/plant.tscn
+++ b/props/plants/plant.tscn
@@ -1,8 +1,8 @@
 [gd_scene format=3 uid="uid://d0k7fuu10fp2p"]
 
 [ext_resource type="Script" uid="uid://c2sr5vbogdqjl" path="res://props/plants/plant.gd" id="1_nxykt"]
-[ext_resource type="Texture2D" uid="uid://pcma1mnvk7u2" path="res://props/sprites/Props.png" id="1_siw38"]
-[ext_resource type="PackedScene" uid="uid://w5gedwn5m48d" path="res://generalnodes/hitbox/hit_box.tscn" id="2_nxykt"]
+[ext_resource type="Texture2D" uid="uid://pcma1mnvk7u2" path="res://props/art/Props.png" id="1_siw38"]
+[ext_resource type="PackedScene" uid="uid://w5gedwn5m48d" path="res://core/combat/hit_box.tscn" id="2_nxykt"]
 
 [sub_resource type="RectangleShape2D" id="RectangleShape2D_nxykt"]
 size = Vector2(31, 29)
diff --git a/ui/hud/party_hud/ability_button.tscn b/ui/hud/party_hud/ability_button.tscn
index c9781bd..b99f771 100644
--- a/ui/hud/party_hud/ability_button.tscn
+++ b/ui/hud/party_hud/ability_button.tscn
@@ -1,7 +1,7 @@
 [gd_scene format=3 uid="uid://c74uo8q80syy"]
 
-[ext_resource type="Script" uid="uid://dky1lgng0nlqo" path="res://GUI/party_hud/ability_button.gd" id="1_ability"]
-[ext_resource type="Script" uid="uid://dlbw0ne58s430" path="res://GUI/party_hud/cooldown_ring.gd" id="2_ring"]
+[ext_resource type="Script" uid="uid://dky1lgng0nlqo" path="res://ui/hud/party_hud/ability_button.gd" id="1_ability"]
+[ext_resource type="Script" uid="uid://dlbw0ne58s430" path="res://ui/hud/party_hud/cooldown_ring.gd" id="2_ring"]
 [ext_resource type="Texture2D" uid="uid://yxhovo54sndw" path="res://icon.svg" id="3_dyyws"]
 
 [node name="AbilityButton" type="Control" unique_id=652290695]
diff --git a/ui/hud/party_hud/party_slot.tscn b/ui/hud/party_hud/party_slot.tscn
index 766e43e..20bb38a 100644
--- a/ui/hud/party_hud/party_slot.tscn
+++ b/ui/hud/party_hud/party_slot.tscn
@@ -1,7 +1,7 @@
 [gd_scene format=3 uid="uid://o4kgmpiulbxg"]
 
-[ext_resource type="Script" uid="uid://do262s6chyr2t" path="res://GUI/party_hud/party_slot.gd" id="1_slot"]
-[ext_resource type="Script" uid="uid://dlbw0ne58s430" path="res://GUI/party_hud/cooldown_ring.gd" id="2_ring"]
+[ext_resource type="Script" uid="uid://do262s6chyr2t" path="res://ui/hud/party_hud/party_slot.gd" id="1_slot"]
+[ext_resource type="Script" uid="uid://dlbw0ne58s430" path="res://ui/hud/party_hud/cooldown_ring.gd" id="2_ring"]
 
 [sub_resource type="StyleBoxFlat" id="StyleBoxFlat_bg"]
 bg_color = Color(0, 0, 0, 0.6)
diff --git a/ui/hud/player_hud/heart_gui.tscn b/ui/hud/player_hud/heart_gui.tscn
index 52ea1f2..c498ca3 100644
--- a/ui/hud/player_hud/heart_gui.tscn
+++ b/ui/hud/player_hud/heart_gui.tscn
@@ -1,7 +1,7 @@
 [gd_scene format=3 uid="uid://b0bl06sye66qs"]
 
-[ext_resource type="Script" uid="uid://i3rnx2nl3502" path="res://GUI/player_hud/heart_gui.gd" id="1_cvmal"]
-[ext_resource type="Texture2D" uid="uid://chseolhv4xd5h" path="res://GUI/player_hud/gui-health.png" id="1_fwptx"]
+[ext_resource type="Script" uid="uid://i3rnx2nl3502" path="res://ui/hud/player_hud/heart_gui.gd" id="1_cvmal"]
+[ext_resource type="Texture2D" uid="uid://chseolhv4xd5h" path="res://ui/hud/player_hud/gui-health.png" id="1_fwptx"]
 
 [node name="Heart" type="Control" unique_id=1041869843]
 custom_minimum_size = Vector2(9, 9)
diff --git a/ui/hud/player_hud/player_hud.tscn b/ui/hud/player_hud/player_hud.tscn
index f1b6b76..79ad591 100644
--- a/ui/hud/player_hud/player_hud.tscn
+++ b/ui/hud/player_hud/player_hud.tscn
@@ -1,8 +1,8 @@
 [gd_scene format=3 uid="uid://m1qnxxfxt1w6"]
 
-[ext_resource type="Texture2D" uid="uid://chseolhv4xd5h" path="res://GUI/player_hud/gui-health.png" id="1_6ggty"]
-[ext_resource type="Script" uid="uid://bdppsjtrmbdds" path="res://GUI/player_hud/player_hud.gd" id="1_oghrr"]
-[ext_resource type="PackedScene" uid="uid://b0bl06sye66qs" path="res://GUI/player_hud/heart_gui.tscn" id="3_4b3j6"]
+[ext_resource type="Texture2D" uid="uid://chseolhv4xd5h" path="res://ui/hud/player_hud/gui-health.png" id="1_6ggty"]
+[ext_resource type="Script" uid="uid://bdppsjtrmbdds" path="res://ui/hud/player_hud/player_hud.gd" id="1_oghrr"]
+[ext_resource type="PackedScene" uid="uid://b0bl06sye66qs" path="res://ui/hud/player_hud/heart_gui.tscn" id="3_4b3j6"]
 
 [node name="PlayerHud" type="CanvasLayer" unique_id=1633031051]
 layer = 2
diff --git a/ui/menus/main_menu/main_menu.tscn b/ui/menus/main_menu/main_menu.tscn
index 420caa4..fcd977e 100644
--- a/ui/menus/main_menu/main_menu.tscn
+++ b/ui/menus/main_menu/main_menu.tscn
@@ -1,6 +1,6 @@
 [gd_scene format=3 uid="uid://chottk576xe66"]
 
-[ext_resource type="Script" uid="uid://bc5y067ymnne6" path="res://main_menu.gd" id="1_menu"]
+[ext_resource type="Script" uid="uid://bc5y067ymnne6" path="res://ui/menus/main_menu/main_menu.gd" id="1_menu"]
 
 [node name="MainMenu" type="Control" unique_id=1485649054]
 layout_mode = 3
diff --git a/ui/menus/pause_menu/pause.tscn b/ui/menus/pause_menu/pause.tscn
index 97afafe..8d64e1f 100644
--- a/ui/menus/pause_menu/pause.tscn
+++ b/ui/menus/pause_menu/pause.tscn
@@ -1,6 +1,6 @@
 [gd_scene format=3 uid="uid://8137607nxqpf"]
 
-[ext_resource type="Script" uid="uid://bb6jgs3bm0mri" path="res://GUI/pause_menu/pause.gd" id="1_242t1"]
+[ext_resource type="Script" uid="uid://bb6jgs3bm0mri" path="res://ui/menus/pause_menu/pause.gd" id="1_242t1"]
 
 [node name="Pause" type="CanvasLayer" unique_id=1399431250]
 process_mode = 3
diff --git a/ui/transition/level_transition.tscn b/ui/transition/level_transition.tscn
index d0a0343..24d05e1 100644
--- a/ui/transition/level_transition.tscn
+++ b/ui/transition/level_transition.tscn
@@ -1,6 +1,6 @@
 [gd_scene format=3 uid="uid://d0l0smnegiw30"]
 
-[ext_resource type="Script" uid="uid://ovysy1dh06tv" path="res://generalnodes/level_transition.gd" id="1_vg6uv"]
+[ext_resource type="Script" uid="uid://ovysy1dh06tv" path="res://ui/transition/level_transition.gd" id="1_vg6uv"]
 
 [sub_resource type="RectangleShape2D" id="RectangleShape2D_xkg8x"]
 resource_local_to_scene = true
diff --git a/ui/transition/transition_screen.tscn b/ui/transition/transition_screen.tscn
index 24f41ac..519c6fe 100644
--- a/ui/transition/transition_screen.tscn
+++ b/ui/transition/transition_screen.tscn
@@ -1,6 +1,6 @@
 [gd_scene format=3 uid="uid://drnyt783bkvk"]
 
-[ext_resource type="Script" uid="uid://dm4m2iyvi1yb0" path="res://generalnodes/transition_screen.gd" id="1_4svag"]
+[ext_resource type="Script" uid="uid://dm4m2iyvi1yb0" path="res://ui/transition/transition_screen.gd" id="1_4svag"]
 
 [sub_resource type="Animation" id="Animation_kfp2j"]
 length = 0.001
